### PR TITLE
Use std.experimental.allocator and remove use of stdx-allocator

### PR DESCRIPTION
**Pull request by nordlow**
Continues the transition started [here](https://github.com/dlang-community/libdparse/pull/463).

Includes replacement of `IAllocator` with `RCIAllocator` and bumping of minimum version of libdparse to 0.19.3 for the sake of also including the transition of `libdparse` to use `std.experimental.allocator`.

### CODE REVIEW

**Comment by nordlow** [comment]
Green now and ready for review and merge. https://github.com/dlang-community/dsymbol/runs/6918218245?check_suite_focus=true seems unrelated.

File: dub.json [hunk 1]
```diff
@@ -5,9 +5,8 @@
 	"authors": ["Brian Schott"],
 	"targetPath": "build",
 	"targetType": "library",
-	"dependencies": {
+    "dependencies": {
```
Comment (WebFreak001) [inline dub.json:8]: whitespace fail

```suggestion
	"dependencies": {
```

Comment (nordlow) [reply dub.json]: Fixed.

**Review by WebFreak001 — APPROVED** [review]

**Comment by WebFreak001** [comment]
new libdparse release is tagged, feel free to update here

**Comment by nordlow** [comment]
> new libdparse release is tagged, feel free to update here

Done via https://github.com/dlang-community/dsymbol/pull/177/commits/c887a4c08e0e5b8ab764ba0764b827b91025f5ed.

**Comment by nordlow** [comment]
Will fix thanks.

**Comment by nordlow** [comment]
> DCD no longer compile with latest version of libdparse/dsymbol
> 
> Maybe it is time to finally merge them all?
> 
> [dlang-community/DCD#669](https://github.com/dlang-community/DCD/issues/669)

I was on to it._

**Comment by nordlow** [comment]
I guess we should tag a dsymbol minor version aswell, @ryuukk.

**Comment by nordlow** [comment]
> I think we should just proceed by removing DSymbol as a dub dependency from DCD and start from there, put DSymbol in the DCD source folder (DCD/src/dsymbol), we'll need to also notice the change in the DSymbol repo to warn users that the development is now happening on the DCD repo instead, and that should be good enough?

So we should just copy the dsymbol tree into DCD, you're saying?

**Comment by WebFreak001** [comment]
PR is here, merging once I resolved the allocator issues on ubuntu CI: https://github.com/dlang-community/DCD/pull/673

**Pull request merged**